src/admin/content-block/components/ContentBlockForm/ContentBlockForm.tsx

```diff
@@ -67,64 +79,70 @@ const ContentBlockForm: FunctionComponent<ContentBlockFormProps> = ({
-					value: defaultProps.options.find(
-						(opt: SelectOption) => opt.value === formState.backgroundColor
-					),
-				};
-				break;
-			case ContentBlockEditor.WYSIWYG:
-				editorProps = {
-					data: formState[fieldKey],
-					onChange: (value: any) => handleChange(fieldKey, value),
-				};
-				break;
-			default:
-				editorProps = {
-					onChange: (value: any) => handleChange(fieldKey, value),
-					value: formState[fieldKey],
-				};
-				break;
-		}
-
-		return <EditorComponent {...defaultProps} {...editorProps} />;
+	const renderFormGroup = (
+		blockType: ContentBlockType,
+		formGroup: ContentBlockComponentsConfig | ContentBlockBlockConfig,
+		formGroupState: ContentBlockComponentState | ContentBlockState,
+		formGroupType: ContentBlockStateType,
+		stateIndex?: number
+	) => {
+		return Object.keys(formGroup.fields).map((key: string, index: number) => (
+			<FormGroup
+				key={`${index}-${blockType}-${key}`}
+				label={
+					stateIndex || stateIndex === 0
+						? `${config.components.name} ${stateIndex + 1}: ${formGroup.fields[key].label}`
+						: formGroup.fields[key].label
+				}
+				error={formErrors[key as keyof ContentBlockComponentState | keyof ContentBlockState]}
+			>
+				<ContentBlockFieldEditor
+					block={{ index, config }}
+					fieldKey={key as keyof ContentBlockComponentState | keyof ContentBlockState}
+					field={formGroup.fields[key]}
+					state={formGroupState}
+					type={formGroupType}
+					stateIndex={stateIndex}
+					handleChange={handleChange}
+				/>
+			</FormGroup>
+		));
 	};
 
-	const renderFormGroups = (cb: ContentBlockConfig) => {
-		return (
-			<Accordion
-				title={`${cb.name} (${index}/${length})`}
-				isOpen={isAccordionOpen}
-				onToggle={setIsAccordionOpen}
-			>
-				{Object.keys(cb.fields).map((key: string) => (
-					<FormGroup
-						key={`${index}-${cb.name}-${key}`}
-						label={cb.fields[key].label}
-						error={formErrors[key as keyof ContentBlockFormStates]}
-					>
-						{renderFieldEditor(key as keyof ContentBlockFormStates, cb.fields[key])}
-					</FormGroup>
-				))}
-			</Accordion>
-		);
+	const renderFormGroups = (
+		blockType: ContentBlockType,
+		formGroup: ContentBlockComponentsConfig | ContentBlockBlockConfig,
+		formGroupType: ContentBlockStateType
+	) => {
+		return Array.isArray(formGroup.state)
+			? formGroup.state.map((formGroupState, stateIndex = 0) =>
+					renderFormGroup(blockType, formGroup, formGroupState, formGroupType, stateIndex)
+			  )
+			: renderFormGroup(blockType, formGroup, formGroup.state, formGroupType);
 	};
 
-	return <Form className="c-content-block-form">{renderFormGroups(config)}</Form>;
+	const renderBlockForm = (contentBlock: ContentBlockConfig) => (
+		<Accordion
+			title={`${contentBlock.name} (${index}/${length})`}
+			isOpen={isAccordionOpen}
+			onToggle={setIsAccordionOpen}
+		>
+			{renderFormGroups(contentBlock.block.state.blockType, components, 'components')}
+			{Array.isArray(components.state) && (
```

Comment:
Eventueel nog adhv een extra property (limit) aangeven wat het max. aantal components mag zijn

Reply:
Zal ik doen in aparte PR.